cores/esp8266/WString.cpp

```diff
@@ -161,11 +162,11 @@ unsigned char String::changeBuffer(unsigned int maxStrLen) {
             return 1;
         } else { // if bufptr && !sso()
             // Using bufptr, need to shrink into sso_buff
-            char temp[sizeof(sso_buf)];
+            char temp[sizeof(sso_buf)] = {0};
```

Comment:
I guess copying less may improve it a bit theoretically, but all other calls to `memcpy` regarding the `sso_buf` do copy the entire buffer.
So for the sake of consistency, it may be wise to have them the same.
The reason it wasn't , did trigger me to have a look at that line.

I don't know how much impact the zeroing has at the time of construction (on the stack).

But since we're also setting it to all zeros in the `init()`, it may be a bit more than needed.
N.B. That zero'ing in the `init` is really needed.

Reply:
It seems only setting a string to "" would have any different behavior in this patch as-is, vs. @devyte's which would explicitly write a 0 at maxstrlen.  I think his is correct (but again, I'm having trouble seeing where this actually changes behavior...)